### DISCUSSION TITLE
Obtain grain boundary angle from miller indices of bicrystals

### DISCUSSION THREAD

**Discussion by SagarChandra** [mtex-toolbox/mtex#1067] [Ask Anything] [answered]
Dear All,

Is it possible to obtain the GB misorientation/inclination angle from the miller indices of two grains of the bicrystal. For instance, 
if left grain is oriented [5 4 1], [1 -1 -1] and [-1 2 -3] and right grain is [5 4 -1], [-1 1 -1] and [-1 2 3] along x- y- and z- directions, respectively, is it possible to obtain the GB angle using MTeX?

Thanks in advance!

Regards,
Sagar

**Comment by kilir** [comment]
Hi Sagar,
do you know the 3D grain shape?
Cheers,
Rüdiger

**Comment by SagarChandra** [comment, marked as answer]
Hi Rüdiger,

Thanks for your reply.
Yes, the bicrystals are rectangular micro pillars with sizes of 5 X 5 X 15
cubic microns.

Regards,
Sagar

**Reply by SagarChandra** [reply]
Hi Rüdiger,

Waiting for your kind reply.

**Reply by kilir** [reply]
Hi,
so assuming it is a crystal shape you are talking about, the following should give you an inclination of the planes. 
```
% since you did not mention, wlet's assume your material is cubic
cs = crystalSymmetry('432');
o = orientation.map(Miller(5, 4, 1,cs,'uvw'), xvector, Miller(1, -1, -1,cs,'uvw'), yvector,cs)

% I also assume your cube shaped grains have {001} planes
% these are the normals
p = o * symmetrise(Miller(0,0,1,cs),'unique','antipodal')

% inclination of plane normals relative to z
% here it depends on you defintion of inclination
angle(p,zvector)/degree
```

Cheers,
Rüdiger

**Comment by SagarChandra** [comment]
Thank you very much for your reply, Rüdiger. I will try it with different
values of Miller indices.
Yes, the crystal symmetry is cubic.

Can you please clarify what you mean by your comment (pardon me if it is a
basic question):
"

your cube shaped grains have {001} planes

these are the normals"

**Reply by kilir** [reply]
Hi,
I simply assume there is a coincidence between the grain shape and the crystal structure. If this is not the case, you need additional information, e.g. boundary traces.

Cheers,
Rüdiger

**Reply by SagarChandra** [reply]
Hi Rüdiger,

Thank you once again for your reply and explanation. The lines you wrote work! The question is answered.

Thanks!

Regards,
Sagar
